rust_src/src/buffers.rs

```diff
@@ -225,6 +238,27 @@ impl LispOverlayRef {
     pub fn end(self) -> LispObject {
         LispObject::from_raw(self.end)
     }
+
+    pub fn iter(self) -> LispOverlayIter {
+        LispOverlayIter {
+            current: Some(self),
+        }
+    }
+}
+
+pub struct LispOverlayIter {
+    current: Option<LispOverlayRef>,
+}
+
+impl Iterator for LispOverlayIter {
+    type Item = LispOverlayRef;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.current.map(|ov| {
+            self.current = unsafe { ov.next.as_ref().map(|o| mem::transmute(o)) };
```

Comment:
And here again. `new_from_ptr`.

Also, I think this method would be more idiomatically written with a `match`. It feels odd to use `map` for side-effects.